mne_qt_browser/_pg_figure.py

```diff
@@ -230,6 +230,63 @@ def _get_channel_scaling(widget, ch_type):
     return inv_norm
 
 
+def _calc_chan_type_to_physical(widget, ch_type, units="mm"):
+    """Convert data to physical units."""
+    # Load in some data and measure its range in a timeperiod
+    start, stop = widget.weakmain()._get_start_stop()
+    data, _ = widget.weakmain()._load_data(start, stop)
```

Comment:
@nmarkowitz I resized my window to show 800 (fT/cm)/cm so that the MEG scale bar should be 1cm but it was a bit big. Adjusting the `__init__` of `class ScaleBar` to use a flat instead of rounded cap (to not draw farther than we should) seemed to fix it:
```
        pen = self.mne.mkPen(color="#AA3377", width=5)
        pen.setCapStyle(Qt.FlatCap)
        self.setPen(pen)
```

However, you should not need to know anything about the actual *data values* to know the effective scale on screen. So using `data` at all (and derivatives like `peak_to_peak_data`) like you do here should not be needed. The only thing that should matter are:

1. The factors that are used to scale the data
2. The number of pixels used to / allocated to show those data (which you should get from the viewbox somehow, could be correctly below?), which you can get from the proportion of the viewbox (in pixels) that is dedicated to showing the data

From those two things you should be able to figure out the effective scaling. Unpacking your code later you have (for `mm` at least):
```
        pixels_per_data_unit = pixel_height / data_height
        peak_to_peak_pixels = peak_to_peak_data * pixels_per_data_unit
        dpi = QApplication.primaryScreen().logicalDotsPerInch()
        peak_to_peak_inches = peak_to_peak_pixels / dpi
        peak_to_peak_mm = peak_to_peak_inches * 25.4
        if units == "mm":
            return (
                _get_channel_scaling(widget, ch_type)
                * peak_to_peak_data
                / peak_to_peak_mm
```
which is equivalent to
```
        dpi = QApplication.primaryScreen().logicalDotsPerInch()
        peak_to_peak_mm = peak_to_peak_data * pixel_height / data_height / dpi * 25.4
        if units == "mm":
            return (
                _get_channel_scaling(widget, ch_type)
                * peak_to_peak_data
                / peak_to_peak_mm
            )
```
which in turn is equivalent to
```
        dpi = QApplication.primaryScreen().logicalDotsPerInch()
        if units == "mm":
            return (
                _get_channel_scaling(widget, ch_type)
                / (pixel_height / data_height / dpi * 25.4)
            )
```
I'll push a commit to simplify the calculation once I verify it's correct